src/Microsoft.ML.Dnn/DnnTransform.cs

```diff
@@ -0,0 +1,1737 @@
+
+                            var editor = VBufferEditor.Create(ref dst, (int)tensorSize);
+
+                            DnnUtils.FetchData<T>(tensor.Data<T>(), editor.Values);
+                            dst = editor.Commit();
+                        };
+                        return valuegetter;
+                    }
+                }
+            }
+
+            private void UpdateCacheIfNeeded(long position, ITensorValueGetter[] srcTensorGetters, string[] activeOutputColNames, OutputCache outputCache)
+            {
+                if (outputCache.Position != position)
+                {
+                    bool addToTheBag = false;
+                    if (!_runners.TryTake(out Runner runner))
+                    {
+                        runner = _parent.Runner.CloneRunner();
+                        addToTheBag = true;
+                    }
+                    for (int i = 0; i < _inputColIndices.Length; i++)
+                    {
+                        var inputName = _parent.Inputs[i];
+                        runner.AddInput(i, srcTensorGetters[i].GetTensor());
+                    }
+
+                    var tensors = runner.Run();
+                    Contracts.Assert(tensors.Length > 0);
+
+                    for (int j = 0; j < activeOutputColNames.Length; j++)
+                        outputCache.Outputs[activeOutputColNames[j]] = tensors[j];
+
+                    outputCache.Position = position;
+
+                    if (addToTheBag)
+                        _runners.Add(runner);
+                }
+            }
+
+            private protected override Func<int, bool> GetDependenciesCore(Func<int, bool> activeOutput)
+            {
+                return col => Enumerable.Range(0, _parent.Outputs.Length).Any(i => activeOutput(i)) && _inputColIndices.Any(i => i == col);
+            }
+
+            protected override DataViewSchema.DetachedColumn[] GetOutputColumnsCore()
+            {
+                var info = new DataViewSchema.DetachedColumn[_parent.Outputs.Length];
+                for (int i = 0; i < _parent.Outputs.Length; i++)
+                    info[i] = new DataViewSchema.DetachedColumn(_parent.Outputs[i], _parent.OutputTypes[i], null);
+                return info;
+            }
+        }
+
+        private interface ITensorValueGetter
+        {
+            Tensor GetTensor();
+
+            void BufferTrainingData();
+
+            Tensor GetBufferedBatchTensor();
+        }
+
+        private class TensorValueGetter<T> : ITensorValueGetter
+        {
+            private readonly ValueGetter<T> _srcgetter;
+            private readonly T[] _bufferedData;
+            private readonly Int64[] _bufferedDataLong;
+            private readonly TensorShape _tfShape;
+            private int _position;
+            private readonly bool _keyType;
+
+            public TensorValueGetter(
```

Comment:
InceptionV3 appears in the enum but it is not used or currently implemented yet as part of the ImageClassifierTrainer, right?
It'd be good to have InceptionV3 as soon as possible in addition to ResNet since most of our samples use Inception and it'll be good to be able to compare "apples with apples". If we compare with Resnet the results will be very different, logically.

Reply:
I'm resolving this comment since I just saw you added INCEPTIONV3 to the code over the weekend. 👍